lib/min_heap.rb

```diff
@@ -44,28 +56,59 @@ def to_s
   end
 
   # This method returns true if the heap is empty
-  # Time complexity: ?
-  # Space complexity: ?
+  # Time complexity: O(1)
+  # Space complexity: O(1)
   def empty?
-    raise NotImplementedError, "Method not implemented yet..."
+    return @store.empty?
   end
 
   private
 
   # This helper method takes an index and
   #  moves it up the heap, if it is less than it's parent node.
   #  It could be **very** helpful for the add method.
-  # Time complexity: ?
-  # Space complexity: ?
+  # Time complexity: O(log n)
+  # Space complexity: O(1)
   def heap_up(index)
-    
+    parent_index = (index - 1 ) / 2
+
+    if @store[parent_index].key > @store[index].key
+      swap(parent_index, index)
+
+      while parent_index > 0 
+        return heap_up(parent_index)
+      end
+    end
   end
 
   # This helper method takes an index and 
   #  moves it up the heap if it's smaller
   #  than it's parent node.
+  # Time complexity: O(log n)
+  # Space complexity: O(1)
   def heap_down(index)
```

Comment:
This method works but the space complexity is O(log n) because you're doing recursion.  Gotta remember the system stack.